builder/orka/orka_client.go

```diff
@@ -175,3 +189,49 @@ func (c *RealOrkaClient) waitForImage(ctx context.Context, name string) error {
 		}
 	}
 }
+
+func (c *RealOrkaClient) WaitForPush(ctx context.Context, namespace, name string) error {
+	return RetryOnWatcherErrorWithTimeout(ctx, 1*time.Hour, func(contextWithTimeout context.Context) error {
+		return c.waitForPush(contextWithTimeout, namespace, name)
+	}, 1*time.Second)
+}
+
+func (c *RealOrkaClient) waitForPush(ctx context.Context, namespace, name string) error {
+	matchLabels := client.MatchingLabels{OrkaJobTypeLabel: OrkaJobTypeRegistryPushValue}
+	if len(name) > 0 {
+		matchLabels[batchv1.JobNameLabel] = name
+	}
+
+	pods := &corev1.PodList{}
+	watcher, err := c.Watch(ctx, pods, client.InNamespace(namespace), matchLabels)
+	if err != nil {
+		return fmt.Errorf("watcher failed to initilize: %s", err)
+	}
+
+	defer watcher.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return WatcherError{Err: errors.New("watcher closed unexpectedly")}
```

Comment:
This is somewhat `nit` - but we can create a constant for this and reuse it (Sonar is complaining about it :))

Something like `var WatcherClosed = errors.New("watcher closed unexpectedly")`

This is a common Go idiom